xrft/xrft.py

```
@@ -234,6 +270,9 @@ def dft(da, spacing_tol=1e-3, dim=None, shift=True, detrend=None, window=False):
     if detrend == 'constant':
         da = da - da.mean(dim=dim)
     elif detrend == 'linear':
+        for d in da.dims:
+            if d not in dim:
+                da = da.chunk({d:1})
```

Can you help me understand why you need to do this?

---

The linear algebra for detrending currently can only take one chunk at a time.

---

But why does dask have to be involved? Are we using map_blocks to apply the detrending?

I'm worried about this. It is generally not a good idea to silently rechunk the user's data. What if this accidentally creates a billion chunks. I would prefer to raise an error and make the user do it.

---

Yes, we use map_blocks.

Ok, I'll make it raise an error but modify it so that it'll automatically chunk the `_segment` dimensions.